### PR TITLE
Adds TEG to roundstart engines

### PR DESCRIPTION
**Pull request by patpol4**
### Intent of your Pull Request
Adds a TEG to the list of possible engines that can spawn along with singulo/tesla and sm
Looks like this:
image

Basic guide to the current setup in comments below

The cold loop is just the SM space loop, the hot loop includes a burn chamber.

Intentionally made it bad with pressure pumps and co2 cans so it's not too powerful without tweaking.
I did test it by myself and stock it runs under 1MW and slowly drops to 0 over the course of around 30-40 minutes

### Why is this change good for the game?
More engi gameplay and popularisation of a rarely used power source, just wait until atmos nerds hear about this.

# Wiki Documentation

### Briefly describe your PR and the impacts of it, in layman's terms. 
Adds Roundstart TEG as a possible engine

### What should players be aware of when it comes to the changes your PR is implementing?
Adds a Roundstart TEG as a possible engine, probably needs a wiki article about it now

### What general grouping does this PR fall under? 
Engineering

### Are there any aspects of the PR that you would like us not to mention on the Wiki?
Not really

# Changelog

:cl:  
rscadd: adds roundstart TEG as a possible engine
/:cl:

### CODE REVIEW

**Comment by patpol4** [comment]
Requesting testmerge @SomeguyManperson

**Comment by maxion12345** [comment]
until teg is tweaked to not be stupid op, this shouldn't be a roundstart engine

**Comment by alexkar598** [comment]
TEG is awful and breaks all the time.


Also im not convinced this is sustainable for very long before it runs out of plasma and heat.

**Comment by Moggram** [comment]
> 
> 
> TEG is awful and breaks all the time.
> 
> Also im not convinced this is sustainable for very long before it runs out of plasma and heat.

it is at low heat or at fusion, but everyone is going try to do a hell burn.

**Comment by patpol4** [comment]
Small changes introduced and some fixes applied
image

**Comment by patpol4** [comment]
Quick guide on basic setup for clueless people who might get directed to this PR page during the testmerge:

1. Wrench all cans
2. Build the TEG
3. Press the button above the one marked with a rectangle
3. Turn all of these to max and on, filters and mixers as noted.
vivaldi_p0iMgSVZj8
4. press the button that is marked with a rectangle in the image above

TEG should now be working at low efficiency, if it turns off after a while, turning the valve that is highlithed with a triangle on for around 10 seconds or more, should help.

**Comment by Cartlord** [comment]
I think my only real issue with this is having it be an Engi thing, I think that if someone wants to screw with gas they'll play Atmos Tech

**Comment by patpol4** [comment]
> I think my only real issue with this is having it be an Engi thing, I think that if someone wants to screw with gas they'll play Atmos Tech

What about the SM? Is that an atmos tech thing aswell?

**Comment by Cartlord** [comment]
I'd say less-so, because of a few things - for one, its roundstart setup isn't set to burn out in 40 minutes like this seems to be; for another, good SM setups seem to mainly revolve around just how much energy a given gas produces, while this relies on creating a temp DIFFERENCE which seems like a much more complex thing with more experimentation that can happen.

**Review by patpol4 — COMMENTED** [review]
I feel like this is now fine to merge.

**Comment by Cartlord** [comment]
Well I played with it, and it's the worst setup I've dealt with in memory (admittedly that isn't sying much). There's no manual, there's no documentation, and there's barely any indication of what to do with this thing. I tried to set it up with the help of another Engineer AND an Engi-Borg, and we got to about here. I can't tell, at this point, why the engine is not working to produce power; I also want to note there's absolutely nothing indicating the TEG itself needs wrenched, I just had to figure that out.
TEGBroken
Also, for some reason the computer delinked from the burn chamber, and Reconnecting it to "teghotchamber" didn't help at all.

**Comment by patpol4** [comment]
> Well I played with it, and it's the worst setup I've dealt with in memory (admittedly that isn't sying much). There's no manual, there's no documentation, and there's barely any indication of what to do with this thing. 

Use the simple guide in this pr comments, if you don't know what you're doing, there is no wiki page for this setup yet.

> I can't tell, at this point, why the engine is not working to produce power; 

your input pressure is too high, follow the guide in the comments, turn on the triangle valve

> I also want to note there's absolutely nothing indicating the TEG itself needs wrenched, I just had to figure that out.

General TEG instructions are already on the wiki

> Also, for some reason the computer delinked from the burn chamber, and Reconnecting it to "teghotchamber" didn't help at all.

Will look into that

**Comment by Cartlord** [comment]
The issue lies in assuming people will come to this PR page. Why would they think to? It's not testmerged, it's full-merged, so there's no easy way in-game to get to this page and I doubt people will think to go to the GitHub repo and search for "TEG" while also making sure to include merged PRs - when the wiki exists, and the page for this engine there is crap.
I understand there is this basic guide, but I feel the need to ask: Why assume people will be able to FIND this basic guide?

**Comment by patpol4** [comment]
Oh im not. Just letting you know since you asked. The wiki page still needs to be made

**Comment by Cartlord** [comment]
BTW you were right about the pressure differential, I just didn't realize it was possible for the differential to be so great the engine didn't output _anything_. Venting most of the hot loop helped.

**Comment by Cartlord** [comment]
New issue I found, uh... you can't get the burn going unless you're an Engineer, which is an issue on lowpop when there ARE no Engis. Possible solution is to make the buttons to open & close burn chamber vents accessible to anyone, since anyone who gets in that room likely has Engi access anyways.

**Comment by patpol4** [comment]
> New issue I found, uh... you can't get the burn going unless you're an Engineer, which is an issue on lowpop when there ARE no Engis. Possible solution is to make the buttons to open & close burn chamber vents accessible to anyone, since anyone who gets in that room likely has Engi access anyways.

We don't really want random tiders being able to vent the burn do we. 
Better solution would probably be making the chamber start closed instead of open.

**Comment by Cartlord** [comment]
> 
> 
> > I think my only real issue with this is having it be an Engi thing, I think that if someone wants to screw with gas they'll play Atmos Tech
> 
> What about the SM? Is that an atmos tech thing aswell?

Re: this, I'd say there're some notable differences between this and the supermatter. For one thing, you can max every pump on the Supermatter engine and be good to go, while doing that here will screw you because it'll lead to the hot loop being massively overpressurized and the engine basically stalling as it stops pumping gas. I've seen a TEG work well, I've seen people produce massive amounts of power with them - but all the best TEG setups I found were entirely contained within the Construction Area. I don't think this works well as a roundstart engine because it's hell to set up, there's nothing in-game explaining how it works, its wiki page is still outdated, and few-if-any people will know to look for the setup guide presented here.
In my experience asking people in OOC, the people who like the TEG are massively outnumbered by those who hate it because of some of the reasons I laid out; I personally think it'd be best for everyone if this was just something you could order at Cargo, and honestly I'm considering making a PR to that effect if I can ever wrap my head around coding.
